Java/src/main/java/org/kondle/socketflow/SocketFlow.java

```diff
@@ -0,0 +1,70 @@
+package org.kondle.socketflow;
+
+import java.net.Socket;
+
+public class SocketFlow
+{
+    private static int initListenersCount = 5;
+    private static int listenersCountExtendSize = 5;
+
+
+    private int connectListenersRealCnt = 0;
+    private int messageListenersRealCnt = 0;
+    private int disconnectListenersRealCnt = 0;
+
+    private SocketFlowConnectListener[] connectListeners;
+    private SocketFlowMessageListener[] messageListeners;
+    private SocketFlowDisconnectListener[] disconnectListeners;
+
+    private Socket socket;
+    public SocketFlow(Socket socket)
+    {
+        this.socket = socket;
+
+        connectListeners = new SocketFlowConnectListener[initListenersCount];
+        messageListeners = new SocketFlowMessageListener[initListenersCount];
+        disconnectListeners = new SocketFlowDisconnectListener[initListenersCount];
+    }
+
+    void addConnectListener(SocketFlowConnectListener listener)
+    {
+        this.connectListeners[connectListenersRealCnt] = listener;
+
+        if (connectListenersRealCnt + 1 == connectListeners.length)
+        {
+            SocketFlowConnectListener[] newListenersArr = new SocketFlowConnectListener[connectListenersRealCnt + 1 + listenersCountExtendSize];
+            System.arraycopy(connectListeners,0,newListenersArr,0,connectListenersRealCnt + 1);
+            this.connectListeners = newListenersArr;
+        }
+
+        connectListenersRealCnt++;
```

Comment:
It's looks as `List` structure.
But why you doen't use default java structures such as `HashMap` with `Bind` method as in C#?